core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -3460,22 +3465,31 @@ class ViewMapping<
 
     m_impl_handle = handle_type(reinterpret_cast<pointer_type>(record->data()));
 
+    functor_type functor =
+        execution_space_specified
+            ? functor_type(exec_space, (value_type*)m_impl_handle,
+                           m_impl_offset.span(), alloc_name)
+            : functor_type((value_type*)m_impl_handle, m_impl_offset.span(),
+                           alloc_name);
+
+    if (false) {
+      // Make sure the destroy functor gets instantiated.
+      // This avoids "cudaErrorInvalidDeviceFunction"-type errors.
+      functor.destroy_shared_allocation();
```

Comment:
Is this not going to generate warnings about unreachable code from one compiler or another? How about `(void)&functor_type::destroy_shared_allocation`?

Reply:
I would try to deal with it when we run into problems. At least our CI seems to be fine with it.